homeassistant/components/melnor/switch.py

```diff
@@ -21,16 +21,11 @@
 from .models import MelnorDataUpdateCoordinator, MelnorZoneEntity
 
 
-def set_is_watering(valve: Valve, value: bool) -> None:
-    """Set the is_watering state of a valve."""
-    valve.is_watering = value
```

Comment:
This pattern was the impetus for the overall changes. It doesn't scale well and the library should be the abstraction layer that handles this rather than HA.